gsocscrape/spiders/gsoc.py

```diff
@@ -0,0 +1,125 @@
+    start_urls = [
+        'https://summerofcode.withgoogle.com/archive/2017/organizations',
+    ]
+
+    def parse(self, response):
+        home_url = "https://summerofcode.withgoogle.com/archive/2017/organizations/"
+        selector = "//li[contains(.,'{org_name}')]/a/@href".format(org_name = org_name)
+        organization_link = response.xpath(selector)
+        
+        if(organization_link):
+            organization_link = organization_link[0].extract().split("/")[4]
+        else:
+            error = "No GSoC data for organization '{org_name}'".format(org_name = org_name)
+            item =  {
+                        'name': error
+                    }
+            error_data = {}
+            error_data[0] = item
+            error_data = OrderedDict(sorted(error_data.items(), key=lambda t: t[0]))
+            with open(os.path.join('_site', 'org_info.yaml'), 'w') as f:
+                yaml.dump(error_data, f)
+            return
+
+        follow_link = home_url + organization_link
+        yield response.follow(follow_link, self.parse_org)
+
+    def parse_org(self, response):        
+        project_url = "https://summerofcode.withgoogle.com/archive/2017/projects/"
+        technology = {}
+
+        id = response.url.split("/")[-2]
+        org_name = response.css("h3.banner__title::text")[0].extract()
+        org_tagline = response.css("h4.org__tagline::text")[0].extract()
+        org_long_description = response.xpath("//div[@class='org__long-description']")[0].extract()
+        org_technologies = response.xpath(".//div[@class='org__meta']/div[contains(.,"
+                            "'Technologies')]/ul/li/text()").extract()
+         
+        count = 0
+        for tech in org_technologies:
+            technology['%s'% str(count)] = tech
+            count = count + 1
+        
+        item =  {
+                    'id': id,
+                    'name': org_name,
+                    'tagline': org_tagline,
+                    'description': org_long_description,
+                    'technologies': technology
+                }
+        org_data = {}
+        org_data[int(item['id'])] = item
+        org_data = OrderedDict(sorted(org_data.items(), key=lambda t: t[0]))
+        
+        with open(os.path.join('_site', 'org_info.yaml'), 'w') as f:
+            yaml.dump(org_data, f)
+        
+        empty_data = {}
+        # Overwrite any previous data with empty dataset
+        with open(os.path.join('_site', 'project_info.yaml'), 'w') as f:
+            yaml.dump(empty_data, f)
+
+        for res in response.css("a.archive-project-card__link"):
+            link = res.xpath("@href").extract()[0]
+            link = link.split("/")[4]
+            url_project = project_url + link
+            yield response.follow(url_project, self.parse_project)
+
+    def parse_project(self, response):
+        mentors = []
+        org_url = "https://summerofcode.withgoogle.com/archive/2017/organizations/"
```

Comment:
`year` should be a parameter, as we'll also want to import other years, and other orgs will have many years of data to import.